src/main/java/org/fogbowcloud/arrebol/execution/k8s/K8sTaskExecutor.java

```diff
@@ -0,0 +1,166 @@
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.OneToOne;
+import javax.persistence.Transient;
+
+import org.apache.log4j.Logger;
+import org.fogbowcloud.arrebol.execution.TaskExecutionResult;
+import org.fogbowcloud.arrebol.execution.TaskExecutor;
+import org.fogbowcloud.arrebol.execution.TaskExecutionResult.RESULT;
+import org.fogbowcloud.arrebol.execution.k8s.client.DefaultK8sClient;
+import org.fogbowcloud.arrebol.execution.k8s.client.K8sClient;
+import org.fogbowcloud.arrebol.execution.k8s.resource.DefaultK8sClusterResource;
+import org.fogbowcloud.arrebol.execution.k8s.resource.K8sClusterResource;
+import org.fogbowcloud.arrebol.models.command.Command;
+import org.fogbowcloud.arrebol.models.command.CommandState;
+import org.fogbowcloud.arrebol.models.task.RequirementsContants;
+import org.fogbowcloud.arrebol.models.task.Task;
+
+import io.kubernetes.client.openapi.ApiException;
+import io.kubernetes.client.openapi.models.V1Job;
+
+@Entity
+public class K8sTaskExecutor implements TaskExecutor {
+
+	@Id
+	@GeneratedValue(strategy = GenerationType.IDENTITY)
+	private Integer id;
+	@Transient
+	private final Logger LOGGER = Logger.getLogger(K8sTaskExecutor.class);
+	@OneToOne(cascade = CascadeType.ALL, orphanRemoval = true, targetEntity = DefaultK8sClusterResource.class)
+	private K8sClusterResource k8sClusterResource;
+	@OneToOne(cascade = CascadeType.ALL, orphanRemoval = true, targetEntity = DefaultK8sClient.class)
+	private K8sClient k8sClient;
+	@Transient
+	private static final long POOLING_PERIOD_TIME_MS = 5 * 1000;
+
+	public K8sTaskExecutor(K8sClusterResource k8sClusterResource, K8sClient k8sClient) {
+		this.k8sClusterResource = k8sClusterResource;
+		this.k8sClient = k8sClient;
+	}
+
+	public K8sTaskExecutor() {
+	}
+
+	@Override
+	public TaskExecutionResult execute(Task task) {
+		TaskExecutionResult taskExecutionResult = null;
+
+		String jobName = task.getId();
+
+		List<Command> commands = task.getTaskSpec().getCommands();
+		String command = joinCommands(commands);
+
+		Map<String, String> requirements = task.getTaskSpec().getRequirements();
+		String imageId = getImageId(requirements);
+
+		LOGGER.debug("Image id: " + imageId);
+		LOGGER.debug("Command: " + command);
+
+		int tasksListSize = task.getTaskSpec().getCommands().size();
+
+		try {
+			V1Job job = k8sClient.createJob(jobName, imageId, command);
+			boolean jobIsRunning = true;
+
+			while (jobIsRunning) {
+				try {
+					sleep(POOLING_PERIOD_TIME_MS);
+				} catch (InterruptedException e) {
+					LOGGER.error(e.getMessage(), e);
+				}
+
+				job = k8sClient.getJob(jobName);
+				jobIsRunning = job.getStatus().getActive() != null;
+			}
+
+			if (wasSuccessful(job)) {
+				successTask(task);
```

Comment:
I think this method can be better named. The reader should know the method's job by its name..

Reply:
Done